torch/linalg/__init__.py

```diff
@@ -248,42 +268,60 @@
 eig = _add_docstr(_linalg.linalg_eig, r"""
 linalg.eig(input, *, out=None) -> (Tensor, Tensor)
 
-Computes the eigenvalues and eigenvectors of a square
-matrix :attr:`input`, or of each such matrix in a batched :attr:`input`.
+Computes the eigenvalue decomposition of a square matrix or batch of square matrices.
+
+For a square matrix :math:`A`, this is defined as
 
-For a single matrix :attr:`input`, the tensor of eigenvalues `w` and the tensor of eigenvectors
-`V` decompose the :attr:`input` such that `input = V diag(w) V⁻¹`, where `V⁻¹` is the inverse of `V`.
+.. math::
+
+    A = V \operatorname{diag}(L) V^{-1}
+
+This decomposition exists if :math:`A` is diagonalizable (for example, when all its eigenvalues are different).
 
-The eigenvalues are not necessarily sorted.
-For real-valued input the resulting eigenvalues and eigenvectors will always be of complex type.
+The returned decomposition is represented as a namedtuple `(eigenvalues, eigenvectors)`,
+which corresponds to :math:`L`, :math:`V` above. If :attr:`input` is a batch of matrices,
+then `L`, `V` are also batched with the same batch dimensions.
 
-Supports input of float, double, cfloat and cdouble dtypes.
+Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
+The eigenvalues and eigenvectors will always be complex-valued, even if :attr:`input` is real.
```

Comment:
This phrasing makes this sentence appear to be a mathematical statement about eigenvalues and eigenvectors, but I think this is trying to make a statement about the output tensors, so it needs to explicitly relate these ambiguous names to the tensors.

"The output \`eigenvalues\` and \`eigenvectors\` tensors will always be complex, even if :attr:\`input\` is real-valued, since the eigenvalues and eigenvectors of real-valued matrices may be complex."

Reply:
This may also be an opportunity to extend that sentence to explain that PyTorch's dtype behavior is divergent from NumPy's. That behavior might also be so nuanced that it could go in a note.

Reply:
Following a previous suggestion, I edited this as:

> The output tensors `eigenvalues` and `eigenvectors` will always be complex-valued, even when :attr:      `input` is real.

I'll add a note explaining this behaviour, as it's done for Hermitian matrices in `eigh`, where the matrices are always real even if the matrix is complex.